### ISSUE TITLE
xUnit1039 is not generic-aware and fails type matching

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2819]
Given this:

```csharp
public static TheoryData<object?, string> NonNullSuccessData = new();

[Theory]
[MemberData(nameof(NonNullSuccessData))]
public void NonNullSuccessCases<T>(T? expectedValue, string serialization) { }
```

xUnit1039 is raised, but it shouldn't be. Generic type arguments should pretty much always match.

**Comment by bradwilson** [comment]
Should ensure all the new assertions that do type equivalency checks are doing them with generic awareness.

**Comment by bradwilson** [comment]
Available in `1.6.0-pre.2`.
